File: multi/presentation/src/main/java/com/example/presentation/screening/model/ScreeningResponseDTO.java
package com.example.presentation.screening.model;

import com.example.domain.screening.Screening;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScreeningResponseDTO {

    private String movieName;

    private String title;

    private String rating;

    private LocalDateTime releaseDate;

    private String thumbnailUrl;

    private Integer runningTime;

    private String genre;

    private String theaterName;


    private LocalDateTime startTime;


    private LocalDateTime endTime;


    private LocalDateTime screeningDate;


    public static ScreeningResponseDTO of(Screening screening){
        var movie = screening.getMovie();
        var theater = screening.getTheater();
        return ScreeningResponseDTO.builder()
                .movieName(movie.getTitle())
                .title(movie.getTitle())
                .rating(movie.getRating().toString())
                .releaseDate(movie.getReleaseDate())
                .thumbnailUrl(movie.getThumbnailUrl())
                .runningTime(movie.getRunningTime())
                .genre(movie.getGenre().toString())
                .theaterName(theater.getTheaterName())
                .startTime(screening.getStartTime())
                .endTime(screening.getEndTime())
                .screeningDate(screening.getScreeningDate())
                .build();
    }

}

File: module-infrastructure/src/main/kotlin/com/example/redis/movie/in/dto/ScreeningResponseDto.kt
package com.example.redis.movie.`in`.dto

import com.example.redis.movie.Screening
import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class ScreeningResponseDto(

    @JsonProperty(value = "theater_id")
    val theaterId: Long,

    @JsonProperty(value = "theater_name")
    val theaterName: String,

    @JsonProperty(value = "screening_id")
    val screeningId: Long,

    @JsonProperty(value = "start_time")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    val startTime: LocalDateTime,

    @JsonProperty(value = "end_time")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    val endTime: LocalDateTime,

) {
    companion object {
        fun toDto(screening: Screening): ScreeningResponseDto {
            return ScreeningResponseDto(
                theaterId = screening.theater.theaterId,
                theaterName = screening.theater.name,
                screeningId = screening.screeningId,
                startTime = screening.startTime,
                endTime = screening.endTime,
            )
        }
    }
}